functions/src/app.ts

```diff
@@ -0,0 +1,36 @@
+// src/app.ts
+import * as express from 'express';
+import * as bodyParser from 'body-parser';
+import { RegisterRoutes } from '../build/routes';
+
+import * as openapi from '../build/swagger.json';
+
+export const app = express();
+
+// DO NOT CHANGE UNLESS firebase.json record is updated as well.
+const API_PREFIX = 'api';
+
+// Use body parser to read sent json payloads
+app.use(
+  bodyParser.urlencoded({
+    extended: true,
+  })
+);
+app.use(bodyParser.json());
+
+// a slight "hack" to get things to play well in Cloud Functions
+// none of the below is required when running locally.
+if (process.env.NODE_ENV === 'production') {
```

Comment:
briefly, why do we need this?

Reply:
Since the CDN rewrite rule in `firebase.json` sends all requests that start with `/api` but the Express app doesn't have a route that starts with `/api`, we need to remove it or make another router with `app.use('/api', actualRouter)` and I opted for just rewriting instead.

Reply:
But when it's running locally you don't need the `/api` prefix and the rewrite rule works differently.

Reply:
Makes sense, thanks for explaining

Reply:
I've added the blurb into the code as well.